### ISSUE TITLE
Has there been a breaking change since v1.1.0 because users are no longer to login to my app when upgrading to 1.4.0? 

### ISSUE THREAD

**Issue by avenmia** [auth0/auth0-react#224]
**Please do not report security vulnerabilities here**. The [Responsible Disclosure Program](https://auth0.com/whitehat) details the procedure for disclosing security issues.

**Thank you in advance for helping us to improve this library!** Please read through the template below and answer all relevant questions. Your additional work here is greatly appreciated and will help us respond as quickly as possible. For general support or usage questions, use the [Auth0 Community](https://community.auth0.com/) or [Auth0 Support](https://support.auth0.com/). Finally, to avoid duplicates, please search existing Issues before submitting one here.

By submitting an Issue to this repository, you agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Describe the problem

I'm upgrading from version 1.1.0 to 1.4.0 and users are no longer able to login. We have a login button that the user can authenticate with auth0, and after attempting to login (either with username/password or logging in with google) they are redirected back to the login screen. I *think* the issue is that getAccessTokenSilently is not setting the access token correctly anymore.  We have a check if the accessToken is not present it will show a loading page that says the user is logging out (which is what's currently happening). However, from the network tab, I can see the access token is returned. I don't know authentication very well so maybe this is a different access token than what the getAccessTokenSilently function is supposed to get? I tried looking through the code changes between 1.1.0 and 1.4.0, but haven't been able to figure out what changes could affect my repo.  Sorry it's not a lot of information to go on, I'm just looking for what sections might have changed so I can adjust.

Here is the check for an access token  and a screen shot from the network tab after a login with google attempt. The accessTokenError that gets returned in the following snippet is blank.
```
else if (route === "exiting" || !accessToken)
    {
        contents = (<LoadingPage error={accessTokenError} target="logging you out" />);
        disableMenu = true;
    }
```

![image](https://user-images.githubusercontent.com/17712276/113072963-cd8e3f80-918d-11eb-874e-65cc0689e084.png)

### What was the expected behavior?

The user should be logged in and redirected to "window.location.origin" which should be the application.

### Reproduction

> Detail the steps taken to reproduce this error, and whether this issue can be reproduced consistently or if it is intermittent.
> **Note**: If clear, reproducable steps or the smallest sample app demonstrating misbehavior cannot be provided, we may not be able to follow up on this bug report.

I'm not able to give steps to reproduce because the repository is private, but I'm wondering what breaking changes might have occurred and just get pointed in the right direction to fix it.  

**Can the behavior be reproduced using the [React SDK Playground](https://github.com/auth0/auth0-react/blob/master/CONTRIBUTING.md#local-development)?**

If so, provide steps:

> Where applicable, please include:
>
> - The smallest possible sample app that reproduces the undesirable behavior
Here are the props Auth0Provider is being initialized with. I tried adding skipRedirectCallback, but no luck. I also added a bit of code that shows what we're using from the auth0 library.

index.tsx
```
import React from "react";
import ReactDOM from "react-dom";
import { Auth0Provider } from "@auth0/auth0-react";
import "./index.css";
import App from "./App";

ReactDOM.render(
    <React.StrictMode>
        <Auth0Provider
            cacheLocation="localstorage"
            useRefreshTokens={true}
            domain="mydomain.us.auth0.com"
            clientId="myClientId"
            redirectUri={window.location.origin}
            audience="https://mydomain.us.auth0.com/api/v2/"
            scope="read:current_user update:current_user_metadata"
        >
            <App />
        </Auth0Provider>
    </React.StrictMode>,
    document.getElementById("root")
);
```

app.tsx
```
const { isAuthenticated, error: authError, user, getAccessTokenSilently, loginWithRedirect, logout } = useAuth0();

useEffect(() => 
    {
        getAccessTokenSilently({ audience: "https://mydomain.us.auth0.com/api/v2/", scope: "read:current_user" })
            .then(token => setAccessToken(token))
            .catch(err => setAccessTokenError(err.message));
    }, [getAccessTokenSilently]);

const doLogin = useCallback(() =>
    {
        changeRoute(undefined);
        loginWithRedirect();
    }, [loginWithRedirect, changeRoute]);

```

> - Log files (redact/remove sensitive information)
> - Application settings (redact/remove sensitive information)
> - `clientId` and `domain` (these are publicly available)
> - Screenshots

### Environment

> Please provide the following:

- **Version of `auth0-react` used:**
- Updating from 1.1.0 to 1.4.0
- **Which browsers have you tested in?**
- Chrome
- **Which framework are you using, if applicable (Angular, React, etc):**
- React
- **Other modules/plugins/libraries that might be involved:**

**Comment by adamjmcgrath** [comment]
Hi @avenmia - we memoized the `getAccessTokenSilently` calls in #150 - which landed in 1.2.0

It looks like your code relies on `getAccessTokenSilently` changing when auth state changes, which would have been effected by this change.

You should check if the user is authenticated before calling `getAccessTokenSilently`, the new dependent key will cause the useEffect hook to trigger when you wan it to.

```js
useEffect(() => 
    {
        if (!isAuthenticated) return;
        getAccessTokenSilently({ audience: "https://mydomain.us.auth0.com/api/v2/", scope: "read:current_user" })
            .then(token => setAccessToken(token))
            .catch(err => setAccessTokenError(err.message));
    }, [getAccessTokenSilently, isAuthenticated]);
```

**Comment by avenmia** [comment]
@adamjmcgrath Thank you, this fixed it for me!
